thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/EntityType.java

```diff
@@ -0,0 +1,39 @@
+package com.linkedin.thirdeye.rootcause.impl;
+
+import com.linkedin.thirdeye.rootcause.Entity;
+import org.apache.commons.lang.StringUtils;
+
+
+/**
+ * Wrapper class for URN prefix based typing of Entity.
+ */
+public final class EntityType {
+  final String prefix;
+
+  public static String extractPrefix(String urn) {
+    String[] parts = urn.split(":");
+    return parts[0] + ":" + parts[1] + ":";
+  }
+
+  public static String extractPrefix(Entity e) {
+    return extractPrefix(e.getUrn());
+  }
+
+  public String getPrefix() {
+    return prefix;
+  }
+
+  public EntityType(String prefix) {
```

Comment:
EntityType serves the same purpose as an Enum - with the difference that it's extensible by other projects importing ThirdEye as a library.